src/PacksInitializable.sol

```diff
@@ -0,0 +1,857 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.28;
+
+import {UUPSUpgradeable} from "@openzeppelin/contracts/proxy/utils/UUPSUpgradeable.sol";
+import {ReentrancyGuardUpgradeable} from "@openzeppelin/contracts-upgradeable/utils/ReentrancyGuardUpgradeable.sol";
+import {PausableUpgradeable} from "@openzeppelin/contracts-upgradeable/utils/PausableUpgradeable.sol";
+import {MEAccessControlUpgradeable} from "./common/MEAccessControlUpgradeable.sol";
+import {PacksSignatureVerifierUpgradeable} from "./common/PacksSignatureVerifierUpgradeable.sol";
+import {IPRNG} from "./common/interfaces/IPRNG.sol";
+import {TokenRescuer} from "./common/TokenRescuer.sol";
+
+contract PacksInitializable is
+    MEAccessControlUpgradeable,
+    PausableUpgradeable,
+    PacksSignatureVerifierUpgradeable,
+    ReentrancyGuardUpgradeable,
+    UUPSUpgradeable,
+    TokenRescuer
+{
+    IPRNG public PRNG;
+    address payable public fundsReceiver;
+
+    CommitData[] public packs;
+    mapping(bytes32 commitDigest => uint256 commitId) public commitIdByDigest;
+
+    uint256 public treasuryBalance; // The operational balance
+    uint256 public commitBalance; // The open commit balance
+    uint256 public packRevenueBalance; // The pack revenue balance
+
+    // Commits are cancellable after time passes unfulfilled
+    uint256 public constant MIN_COMMIT_CANCELLABLE_TIME = 1 hours;
+    uint256 public commitCancellableTime = 1 days;
+    mapping(uint256 commitId => uint256 cancellableAt) public commitCancellableAt;
+
+    // NFT fulfillment option expires after a short time
+    uint256 public constant MIN_NFT_FULFILLMENT_EXPIRY_TIME = 30 seconds;
+    uint256 public nftFulfillmentExpiryTime = 2 minutes;
+    mapping(uint256 commitId => uint256 expiresAt) public nftFulfillmentExpiresAt;
+
+    bytes32 public constant FUNDS_RECEIVER_MANAGER_ROLE = keccak256("FUNDS_RECEIVER_MANAGER_ROLE");
+
+    mapping(address cosigner => bool active) public isCosigner;
+    mapping(address receiver => uint256 counter) public packCount;
+    mapping(uint256 commitId => bool fulfilled) public isFulfilled;
+    mapping(uint256 commitId => bool cancelled) public isCancelled;
+
+    uint256 public payoutBps; // When user selects payout as reward
+    uint256 public minReward; // Min ETH reward for a commit (whether it's NFT or payout)
+    uint256 public maxReward; // Max ETH reward for a commit (whether it's NFT or payout)
+    uint256 public minPackPrice; // Min ETH pack price for a commit
+    uint256 public maxPackPrice; // Max ETH pack price for a commit
+
+    // Storage gap for future upgrades
+    uint256[50] private __gap;
```

Comment:
For this contract, unless you plan on have child contracts, I dont think you need this. If you were going to import it like

` contract SomethingNew is PacksInitializable {
  uint256 someNumber;
}` 

then the reserved storage gap would allow you to add new variables, adjust the gap, and it would not impact the storage in `SomethingNew`